jwt-bearer-auth-yesod/README.lhs

```diff
@@ -174,41 +186,41 @@ decoded payload for you to look at and decide if the request is allowed.
 Now, you will almost certainly want to actually inspect the fields on the payload when making your
 decision. The `ClaimsSet` type in the `jose` library ONLY supports the claims that are required by
 the RFC for JWTs, but you can extend them with more fields. The supported way to do that is by
-providing your own data type that contains a `ClaimsSet` and provide a lens to focus on that
-`ClaimsSet`. For an example, see `Web.Auth.Bearer.JWT.Claims`. You will also need a `FromJSON`
-instance (you would also need a `ToJSON` instance if you were doing the signing, but we're only
-going to be validating ones that have already been signed.)
+providing your own data type that implements `HasClaimsSet`, providing a lens to focus on that
+`ClaimsSet`. Anything that has an instance of `HasClaimsSet` also has lenses for each individual
+claim: `claimAud`, `claimIss`, `claimSub`, etc. This library provides a general wrapper for this,
+`JWTClaims extraClaims`, where `extraClaims` is whatever claims you want in your app.
+
+Your `extraClaims` needs a `FromJSON` instance; you would also need a `ToJSON` instance if
+you were doing the signing, but we're only going to be validating ones that have already been
+signed. Here is an example of a type that's already provided for you in
+`Web.Auth.Bearer.JWT.Claims`, that adds a single extra claim called `scp` that's a list of strings.
+Use this as a template if you need to implement your own, additional claims.
+
+```haskell ignore
+data ScpClaims = ScpClaims
+  { scp :: [String]
+  }
+  deriving stock (Eq, Show)
```

Comment:
I think this won't compile without `Generic` here.

Maybe it'd be better if you could un-ignore this block, to get coverage on it?